src/libteam/0005-libteam-Add-warm_reboot-mode.patch

```diff
@@ -0,0 +1,354 @@
+diff --git a/libteam/ifinfo.c b/libteam/ifinfo.c
+index 72155ae..266cc21 100644
+--- a/libteam/ifinfo.c
++++ b/libteam/ifinfo.c
+@@ -58,6 +58,7 @@ struct team_ifinfo {
+ 	char			ifname[IFNAMSIZ];
+ 	uint32_t		master_ifindex;
+ 	bool			admin_state;
++        bool                    orig_addr_updated; /* FIXME: Check this. I think we don't need this flag */
+ #define MAX_PHYS_PORT_ID_LEN 32
+ 	char			phys_port_id[MAX_PHYS_PORT_ID_LEN];
+ 	size_t			phys_port_id_len;
+@@ -105,15 +106,17 @@ static void update_hwaddr(struct team_ifinfo *ifinfo, struct rtnl_link *link)
+ 	hwaddr_len = nl_addr_get_len(nl_addr);
+ 	if (ifinfo->hwaddr_len != hwaddr_len) {
+ 		ifinfo->hwaddr_len = hwaddr_len;
+-		if (!ifinfo->master_ifindex)
++		if (!ifinfo->orig_addr_updated)
+ 			ifinfo->orig_hwaddr_len = hwaddr_len;
+ 		set_changed(ifinfo, CHANGED_HWADDR_LEN);
+ 	}
+ 	hwaddr = nl_addr_get_binary_addr(nl_addr);
+ 	if (memcmp(ifinfo->hwaddr, hwaddr, hwaddr_len)) {
+ 		memcpy(ifinfo->hwaddr, hwaddr, hwaddr_len);
+-		if (!ifinfo->master_ifindex)
++		if (!ifinfo->orig_addr_updated) {
+ 			memcpy(ifinfo->orig_hwaddr, hwaddr, hwaddr_len);
++                        ifinfo->orig_addr_updated = true;
++                }
+ 		set_changed(ifinfo, CHANGED_HWADDR);
+ 	}
+ }
+diff --git a/teamd/teamd.c b/teamd/teamd.c
+index c987333..37c2fb3 100644
+--- a/teamd/teamd.c
++++ b/teamd/teamd.c
+@@ -116,7 +116,9 @@ static void print_help(const struct teamd_context *ctx) {
+             "    -D --dbus-enable         Enable D-Bus interface\n"
+             "    -Z --zmq-enable=ADDRESS  Enable ZeroMQ interface\n"
+             "    -U --usock-enable        Enable UNIX domain socket interface\n"
+-            "    -u --usock-disable       Disable UNIX domain socket interface\n",
++            "    -u --usock-disable       Disable UNIX domain socket interface\n"
++            "    -w --warm-reboot         Warm-reboot startup mode\n"
```

Comment:
>warm-reboot [](start = 23, length = 11)

I believe these was discussion on the terminology. Use 'warm-start'? @lguohan

Reply:
agree, warm-start is better

Reply:
fixed